src/ORM/DataObjectSchema.php

```diff
@@ -471,7 +479,13 @@ protected function cacheDatabaseFields($class)
             $fieldObj->setTable($class);
             $nestedFields = $fieldObj->compositeDatabaseFields();
             foreach ($nestedFields as $nestedName => $nestedSpec) {
-                $dbFields["{$fieldName}{$nestedName}"] = $nestedSpec;
+                $compositeName = $fieldName . $nestedName;
+                $dbFields[$compositeName] = $nestedSpec;
+
+                // Handle configurable indexes, e.g. for polymorphic keys
```

Comment:
Yes ideally, but at that point you don't know whether it's a polymorphic key because at this point it gets changed into an Int and a DBClassName instead